### PR TITLE
add io.github.nostar.droidstar

### PR DESCRIPTION
**Pull request by lu9dce**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by lu9dce** [comment]
bot, build io.github.nostar.droidstar

**Comment by lu9dce** [comment]
bot, build  io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
bot, build io.github.nostar.droidstar

File: io.github.nostar.DroidStar.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+modules:
+  - name: imbe_vocoder
+    buildsystem: simple
+    build-commands:
+      - make
+      - install -Dm644 libimbe_vocoder.a /app/lib/libimbe_vocoder.a
+    sources:
+      - type: git
+        url: https://github.com/nostar/imbe_vocoder.git
+        branch: master
```
Comment (hfiguiere) [inline io.github.nostar.DroidStar.yml:30]: as I said in the PR you closed: no branch. You need a commit.

File: io.github.nostar.DroidStar.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  - name: droidstar
+    buildsystem: qmake
+    post-install:
+      - install -Dm644 images/droidstar.png /app/share/icons/hicolor/128x128/apps/io.github.nostar.DroidStar.png
+      - install -Dm644 com.github.nostar.DroidStar.desktop /app/share/applications/io.github.nostar.DroidStar.desktop
+      - appstream-compose --basename=com.github.nostar.DroidStar --prefix=/app --origin=flatpak io.github.nostar.DroidStar
+    sources:
+      - type: git
+        url: https://github.com/nostar/DroidStar.git
+        branch: main
```
Comment (hfiguiere) [inline io.github.nostar.DroidStar.yml:40]: as I said in the PR you closed: no branch. You need a commit.

File: io.github.nostar.DroidStar.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    sources:
+      - type: git
+        url: https://github.com/nostar/imbe_vocoder.git
+        branch: master
+  - name: droidstar
+    buildsystem: qmake
+    post-install:
+      - install -Dm644 images/droidstar.png /app/share/icons/hicolor/128x128/apps/io.github.nostar.DroidStar.png
+      - install -Dm644 com.github.nostar.DroidStar.desktop /app/share/applications/io.github.nostar.DroidStar.desktop
+      - appstream-compose --basename=com.github.nostar.DroidStar --prefix=/app --origin=flatpak io.github.nostar.DroidStar
```
Comment (hfiguiere) [inline io.github.nostar.DroidStar.yml:36]: as I said in the PR you closed: this is not necessary
```suggestion
```

**Comment by lu9dce** [comment]
bot, build io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
@hfiguiere mode.h:26:10: fatal error: imbe_vocoder_api.h: No such file or directory
   26 | #include <imbe_vocoder_api.h>

If I try to compile it on my PC it generates the same .. how to solve that ! what's missing!

File: io.github.nostar.DroidStar.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  - /share/doc
+  - /doc
+  - "*.a"
+  - "*.la"
+modules:
+  - name: imbe_vocoder
+    buildsystem: simple
+    build-commands:
+      - make
+      - install -Dm644 libimbe_vocoder.a /app/lib/libimbe_vocoder.a
```
Comment (AsciiWolf) [inline io.github.nostar.DroidStar.yml:26]: You also need to install the _imbe_vocoder_api.h_ file into _/app/include_.

File: io.github.nostar.DroidStar.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/nostar/imbe_vocoder/archive/refs/heads/master.zip
```
Comment (hfiguiere) [inline io.github.nostar.DroidStar.yml:29]: I will repeat: no branch. Tricking it by downloading the `master` archive doesn't cut it. You need to pin a specific commit or tag.

File: io.github.nostar.DroidStar.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        sha256: c9b55c55650e0d16301dcd2f04edbbd2130f1809aded5fc364f95ff28577a937
+  - name: droidstar
+    buildsystem: qmake
+    post-install:
+      - install -Dm644 images/droidstar.png /app/share/icons/hicolor/128x128/apps/io.github.nostar.DroidStar.png
+      - install -Dm644 com.github.nostar.DroidStar.desktop /app/share/applications/io.github.nostar.DroidStar.desktop
+    sources:
+      - type: archive
+        url: https://github.com/nostar/DroidStar/archive/refs/heads/main.zip
```
Comment (hfiguiere) [inline io.github.nostar.DroidStar.yml:38]: I will repeat: no branch. Tricking it by downloading the `master` archive doesn't cut it. You need to pin a specific commit or tag.

File: io.github.nostar.DroidStar.yml (hunk 4 above)
Comment (hfiguiere) [reply io.github.nostar.DroidStar.yml:26]: there is a make install target that should be used instead.

File: io.github.nostar.DroidStar.yml (hunk 5 above)
Comment (lu9dce) [reply io.github.nostar.DroidStar.yml:31]: I don't know.. A few years ago when I made manifestos for flatpak we didn't use gitbub ... Between some new changes and using GitHub for everything costs me twice as much.. I don't just have to see the changes but try to use GitHub which is complicated

File: io.github.nostar.DroidStar.yml (hunk 6 above)
Comment (lu9dce) [reply io.github.nostar.DroidStar.yml:40]: I will try to solve it .. I have more than 60 amateur radio programs to upload .. Have me patience

File: io.github.nostar.DroidStar.yml (hunk 4 above)
Comment (lu9dce) [reply io.github.nostar.DroidStar.yml:26]: ok ascii !! tks Now comes another error .. Let's go apoco! Je
Glad to see you again.. I do not know if you remember but start with Fldigi !

**Comment by lu9dce** [comment]
bot, build io.github.nostar.DroidStar

File: io.github.nostar.DroidStar.yml (hunk 6 above)
Comment (AsciiWolf) [reply io.github.nostar.DroidStar.yml]: Nice! :-)

File: io.github.nostar.DroidStar.yml (hunk 4 above)
Comment (AsciiWolf) [reply io.github.nostar.DroidStar.yml:18]: Of course I remember you and I am glad to see you back here on GitHub! :-)

File: io.github.nostar.DroidStar.yml (hunk 5 above)
Comment (AsciiWolf) [reply io.github.nostar.DroidStar.yml]: Here on Flathub we have a bot that can automatically create new Pull Requests with updates when a new version of any component of the Flatpak gets released so you don't have to update it manually, just test and merge the PR. :-) See [this](https://github.com/flathub/flatpak-external-data-checker#on-flathub) for more info.

**Comment by lu9dce** [comment]
bot, build io.github.nostar.DroidStar

**Review by hfiguiere — CHANGES_REQUESTED** [review]
packge still use hea of a branch for source, and will break at the next commit upstream.

**Comment by lu9dce** [comment]
well sorry but at the moment I will not continue with flatpak .. Unfortunately I do not understand the operation of GitHub .. I spend more time trying to make it walk GitHub than configured the manifest .. GitHub goes beyond my knowledge.. Even if I have to update JTDX it is too complex.. I am already 60 years old and the head does not give me so much .. and every text in English I have to translate and vis .. I would really like to collaborate with Flatpak

**Comment by lu9dce** [comment]
I will try to create packages locally for my own use or if I can learn how to make some repo .. I will also try to finish with DroidStar to see if I find any solution

**Comment by lu9dce** [comment]
bot, compilar io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
bot, build io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
bot, build io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
bot, build io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
Captura desde 2023-04-04 23-29-53

**Comment by lu9dce** [comment]
bot, build io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
> > ```
> > * [x]  I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
> > 
> > * [x]  I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
> > ```


I don't know what it refers to.. I do not know what are the tildes that gives to put .. I don't know anything about English and I have to use a translator.. Mark all the dots because I don't know what they say.. just that

**Comment by lu9dce** [comment]
Try translating line by line and use the AI to tell me what it refers to.. I think that as well as this marked now would be normal .. Excuse me.. Online translators do not reflect the event

**Review by AsciiWolf — CHANGES_REQUESTED** [review]

File: io.github.nostar.DroidStar.desktop [hunk 7]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Exec=DroidStar
+Icon=io.github.nostar.DroidStar
+Name=DroidStar
```
Comment (AsciiWolf) [inline io.github.nostar.DroidStar.desktop:4]:
```suggestion
Name=DroidStar
GenericName=D-Star client
Comment=A D-Star client for PC
```

**Review by AsciiWolf — CHANGES_REQUESTED** [review]

File: io.github.nostar.DroidStar.yml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        commit: 03423cb00b8e9ba1be164493e8b98b5bd7c1508c
+  - name: DroidStar
+    buildsystem: simple
+    build-commands:
+      - qmake LIBS+=-L$FLATPAK_DEST/lib DroidStar.pro
+      - make -j`nproc`
+    post-install:
+      - install -D io.github.nostar.DroidStar.metainfo.xml $FLATPAK_DEST/share/metainfo/io.github.nostar.DroidStar.metainfo.xml
+      - install -D DroidStar $FLATPAK_DEST/bin/DroidStar
+      - install -D images/droidstar.png $FLATPAK_DEST/share/icons/hicolor/128x128/apps/io.github.nostar.DroidStar.png
```
Comment (AsciiWolf) [inline io.github.nostar.DroidStar.yml:34]: Upstream icon seems to be 423px × 423px. You first have to convert the icon to 128px x 128px (or 256px x 256px).

**Review by AsciiWolf — CHANGES_REQUESTED** [review]

File: io.github.nostar.DroidStar.yml (hunk 8 above)
Comment (lu9dce) [reply io.github.nostar.DroidStar.yml]: ok

**Comment by AsciiWolf** [comment]
GridTracker would be great. :-)

**Comment by AsciiWolf** [comment]
bot, build io.github.nostar.droidstar

**Comment by AsciiWolf** [comment]
bot, build io.github.nostar.DroidStar

**Comment by lu9dce** [comment]
when droidstar is approved.. I start with GridTracker .. I will try this time to make as few mistakes as possible!

**Comment by AsciiWolf** [comment]
@lu9dce Any chance you could prepare the GridTracker Flatpak? Thanks! :-)

**Pull request closed without merging**